src/endpoint.cc

```diff
@@ -844,7 +843,9 @@ worker endpoint::do_subscribe(filter_type&& filter,
   BROKER_ASSERT(sink != nullptr);
   using caf::async::make_spsc_buffer_resource;
   // Get a pair of connected resources.
-  auto [con_res, prod_res] = make_spsc_buffer_resource<data_message>();
+  // Note: structured bindings with values confuses clang-tidy's leak checker.
+  auto resources = make_spsc_buffer_resource<data_message>();
+  auto& [con_res, prod_res] = resources;
```

Comment:
Concerning the old code:

```c++
auto [con_res, prod_res] = make_spsc_buffer_resource<data_message>();
```

This deconstructs the temporary object into two values. The compiler should break this down into something like this:

```c++
auto tmp = make_spsc_buffer_resource<data_message>();
auto con_res = std::move(tmp.first);
auto prod_res = std::move(tmp.second);
```

For some reason, `clang-format` gets confused with object lifetimes for the code expanded from the structured binding and complains about a leak that doesn't happen. The new code simply makes the temporary variable visible and skips the moves by simply pointing back into the (now named) temporary object. The new version should expand to something like this:

```c++
auto resources = make_spsc_buffer_resource<data_message>();
auto& con_res = resources.first;
auto& prod_res = resources.second;
```

From a coding perspective, there is no functional change here - other than the temporary object got put on the stack and has a name now. I'd prefer the old code (simply to have a one line less of code) if it wouldn't confuse `clang-format`.

So yes, having references now is intentional to avoid extra copies. We simply need to place the result of `make_spsc_buffer_resource` on the stack for "un-confusing" `clang-format` regarding the object lifetimes.